nav2_costmap_2d/include/nav2_costmap_2d/costmap_filters/binary_filter.hpp

```diff
@@ -0,0 +1,152 @@
+/*********************************************************************
+ *
+ * Software License Agreement (BSD License)
```

Comment:
Apache?

Reply:
Binary filter source was taken from [speed_filter.cpp](https://github.com/ros-planning/navigation2/blob/main/nav2_costmap_2d/plugins/costmap_filters/speed_filter.cpp) and [keepout_filter.cpp](https://github.com/ros-planning/navigation2/blob/main/nav2_costmap_2d/plugins/costmap_filters/keepout_filter.cpp). Both of these sources are being a part of `Costmap2D` and having a BSD-3 license. Therefore, I think we should distribute all derivative works under the same license.

The opposite situation for the testcases, that were developed "from the scratch", so they having more actual Apache-2.0 license.